dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -132,6 +132,9 @@
 		<StartupHookSupport Condition="'$(StartupHookSupport)' == ''">false</StartupHookSupport>
 		<UseSystemResourceKeys Condition="'$(UseSystemResourceKeys)' == ''">true</UseSystemResourceKeys>
 		<UseNativeHttpHandler Condition="'$(_PlatformName)' != 'macOS' And '$(UseNativeHttpHandler)' == ''">true</UseNativeHttpHandler>
+
+		<!-- We don't need to generate reference assemblies for apps or app extensions -->
+		<ProduceReferenceAssembly Condition="'$(ProduceReferenceAssembly)' == '' And ('$(OutputType)' == 'Exe' Or '$(IsAppExtension)' == 'true')">false</ProduceReferenceAssembly>
```

Comment:
hmm... iirc we're using `.dll` for the main assembly for net6 so that would never work (except for app extensions)

Reply:
We use `OutputType=Exe`: 

https://github.com/xamarin/xamarin-macios/blob/9d97d43f74c28fa4e0c1cc9fa70a0983f24b7ad9/tests/dotnet/MySingleView/MySingleView.csproj#L6

We also depend on this elsewhere in this exact file: 

https://github.com/xamarin/xamarin-macios/blob/2aaa7a31b2313dc7bcd50af04c85b4db6d2c4ab0/dotnet/targets/Xamarin.Shared.Sdk.targets#L15-L35

Reply:
hmm... is that recent ?

because size comparison shows `MySingleView.dll` for .net

or maybe the issue is elsewhere... `<OutputType>` not doing what seems logical ?

Reply:
> hmm... is that recent ?

No, it's not recent, it's always been this way.

> 
> because size comparison shows `MySingleView.dll` for .net
> 
> or maybe the issue is elsewhere... `<OutputType>` not doing what seems logical ?

Yes, OutputType is doing something different now. I'm not quite sure what, but I believe it doesn't affect the extension for the output assembly anymore.